### PR TITLE
make sure /etc/localtime issues don't cause systemd to fail boot

### PR DESCRIPTION
**Pull request by poettering**
Not really a fix for #9602, but still addresses it somewhat

### CODE REVIEW

File: src/core/manager.c [hunk 1]
```diff
@@ -839,9 +839,7 @@ int manager_new(UnitFileScope scope, ManagerTestRunFlags test_run_flags, Manager
                 if (r < 0)
                         return r;
 
-                r = manager_setup_timezone_change(m);
-                if (r < 0)
-                        return r;
+                (void) manager_setup_timezone_change(m);
```
Comment (yuwata) [inline src/core/manager.c:842]: Instead of voidify this, what do you think about adding taint string?

Comment (poettering) [reply src/core/manager.c:842]: well, i think taint is something we can do for stuff that is the user's/admin's/vendor's fault I think. i.e. recognizable ways how people are misusing things. But the above is not of that kind, manager_setup_timezone_change() is not supposed to fail, and if it fails then something is weird (and we should log about it) but very likely it's our fault not the user's hence we shouldn't taint the system I think...

Comment (yuwata) [reply src/core/manager.c:842]: Thank you for the explanation about our stance to the taint.

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**